projects/smoke_adt7476a/src/main.c

```diff
@@ -0,0 +1,76 @@
+#include "adt7476a_fan_controller.h"
+#include "adt7476a_fan_controller_defs.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define SET_SPEED_INTERVAL 5
+#define FAN_SPEED_INCREMENT 10
+#define I2C_WRITE_ADDR_1 0x5E
+#define I2C_WRITE_ADDR_2 0x58
+#define I2C_READ_ADDR_1 0x5F
+#define I2C_READ_ADDR_2 0x59
+
+// 2 storages for 2 components
+static Adt7476aStorage s_storage_1;
+static Adt7476aStorage s_storage_2;
+int s_current_speed;
+
+static void prv_periodic_set_speed(SoftTimerId id, void *context) {
+  s_current_speed += FAN_SPEED_INCREMENT;
+  s_current_speed = s_current_speed % 101;
```

Comment:
why not just 100? I feel like I'm missing something here (which may be my intro programming memories)

Reply:
wouldn't we not be able to hit full speed if it was `% 100`?

Reply:
im dumb